app/services/fdc_offer_broker.rb

```diff
@@ -2,13 +2,23 @@
 
 # Finds wholesale offers for retail products.
 class FdcOfferBroker
+  Solution = Struct.new(:product, :factor, :offer)
```

Comment:
If we were on ruby >=3.2 I would have suggested to use `Data` instead of `Struct` here : https://ruby-doc.org/3.2.5/Data.html